File: sem3/map/ToyLanguageInterpreter/src/model/VarExpr.java
package model;

import utils.SymbolTable;

/**
 * Created by mirko on 12/10/2016.
 */
public class VarExpr implements Expression {
    private String name;

    public VarExpr(String name) {
        this.name = name;
    }

    @Override
    public int evaluate(SymbolTable<String, Integer> symTable) {
        return symTable.getValue(name);
    }

    @Override
    public String toString() {
        return name;
    }
}

File: sem3/map/ToySharpLanguageInterpreter/ToySharpLanguageInterpreter/src/model/expressions/VarExpr.cs
using System;

namespace ToySharpLanguageInterpreter
{
	public class VarExpr : Expression
	{
		private string name;

		public VarExpr(string name)
		{
			this.name = name;
		}

		public int evaluate(SymbolTable<string, int> symTable)
		{
			return symTable.getValue(name);
		}

		public override string ToString()
		{
			return name;
		}
	}
}